test_helpers/main.go

```diff
@@ -0,0 +1,126 @@
+package test_helpers
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/tarantool/go-tarantool"
+)
+
+type StartOpts struct {
+	Command      string
+	Server       string
+	WorkDir      string
+	User         string
+	Pass         string
+	WaitStart    time.Duration
+	ConnectRetry uint
+	RetryTimeout time.Duration
+}
+
+func InstallRock(rock string, version string) error {
+	cmd := exec.Command("tarantoolctl", "rocks", "install", rock, version)
+
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isReady(server string, opts *tarantool.Opts) error {
+	var err error
+	var conn *tarantool.Connection
+	var resp *tarantool.Response
+
+	conn, err = tarantool.Connect(server, *opts)
+	if err != nil {
+		return err
+	}
+	if conn == nil {
+		return errors.New("Conn is nil after connect")
+	}
+	defer conn.Close()
+
+	resp, err = conn.Ping()
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return errors.New("Response is nil after ping")
+	}
+
+	return nil
+}
+
+func StartTarantool(startOpts StartOpts) (*exec.Cmd, error) {
+	var err error
+	var cmd *exec.Cmd
+
+	// Prepare tarantool command
+	cmd = exec.Command("tarantool", "config.lua")
+
+	cmd.Env = append(
+		os.Environ(),
+		fmt.Sprintf("TARANTOOL_WORK_DIR=%s", startOpts.WorkDir),
+		fmt.Sprintf("TARANTOOL_LISTEN=%s", startOpts.Server),
+	)
+
+	// Clean up existing work_dir.
+	err = os.RemoveAll(startOpts.WorkDir)
+	if err != nil {
+		return cmd, err
+	}
+
+	// Create work_dir.
+	err = os.Mkdir(startOpts.WorkDir, 0755)
+	if err != nil {
+		return cmd, err
+	}
+
+	// Start tarantool.
+	err = cmd.Start()
+	if err != nil {
+		return cmd, err
+	}
+
+	// Try to connect and ping tarantool.
+	// Using reconnect opts do not help on Connect,
+	// see https://github.com/tarantool/go-tarantool/issues/136
+	time.Sleep(startOpts.WaitStart)
+
+	opts := tarantool.Opts{
+		Timeout:    500 * time.Millisecond,
+		User:       startOpts.User,
+		Pass:       startOpts.Pass,
+		SkipSchema: true,
+	}
+
+	var i uint
+	for i = 0; i <= startOpts.ConnectRetry; i++ {
+		err = isReady(startOpts.Server, &opts)
+
+		// Both connect and ping is ok.
+		if err == nil {
+			break
+		}
+
+		if i != startOpts.ConnectRetry {
+			time.Sleep(startOpts.RetryTimeout)
+		}
+	}
+
+	return cmd, err
+}
+
+func StopTarantool(cmd *exec.Cmd) {
+	if cmd == nil || cmd.Process == nil {
+		return
+	}
+
+	cmd.Process.Kill()
```

Comment:
It may be worth waiting for the process to be terminated.

Reply:
Fixed